samples/quickperf/src/main/java/com/google/cloud/jdbc/quickperf/QuickPerf.java

```diff
@@ -0,0 +1,497 @@
+                    return;
+                }
+
+                ResultSet rs = statement.getResultSet();
+                while (rs.next()) {
+                    String value = rs.getString(1);
+                    sampledValueList.add(value);
+                }
+
+                System.out.println(String.format("Finished sampling %s records", sampledValueList.size()));
+            } catch (SQLException e) {
+                e.printStackTrace();
+            }
+
+        } catch (SQLException e) {
+            e.printStackTrace();
+        }
+    }
+
+    private String createConnectionURL(Config config) {
+        if (config.isIsEmulator()) {
+            return String.format(
+                    "jdbc:cloudspanner:/projects/%s/instances/%s/databases/%s?autoConfigEmulator=true",
+                    config.getProject(), config.getInstance(), config.getDatabase());
+        } else {
+            return String.format(
+                    "jdbc:cloudspanner:/projects/%s/instances/%s/databases/%s",
+                    config.getProject(), config.getInstance(), config.getDatabase());
+        }
+    }
+
+    public void run() {
+        if (config.getBatchSize() > 0) {
+            int val = (int) Math.ceil((double) config.getIterations() / config.getBatchSize());
+            measures = new float[val];
+        } else {
+            measures = new float[config.getIterations()];
+        }
+
+        String connectionUrl = createConnectionURL(config);
+
+        try {
+            Connection connection = DriverManager.getConnection(connectionUrl);
+
+            // determin database dialect to set right tagging syntax
+            boolean isGoogleSQL;
+            if (connection.unwrap(CloudSpannerJdbcConnection.class).getDialect()
+                    .equals(Dialect.GOOGLE_STANDARD_SQL)) {
+                isGoogleSQL = true;
+            } else {
+                isGoogleSQL = false;
+            }
+
+            connection.setAutoCommit(false);
+
+            // if there is DML switch to r/w transaction mode and apply transaction tagging.
+            // Otherwise set to read-only mode.
+            if (config.getQuery().contains("INSERT") ||
+                    config.getQuery().contains("UPDATE")
+                    || config.getQuery().contains("DELETE")) {
+                // read-write
+                connection.createStatement().execute("SET TRANSACTION READ WRITE");
+
+                if (isGoogleSQL) {
+                    connection.createStatement().execute(String.format("SET TRANSACTION_TAG = '%s'", DEFAULT_TAG));
+                } else {
+                    connection.createStatement()
+                            .execute(String.format("SET SPANNER.TRANSACTION_TAG = '%s'", DEFAULT_TAG));
+                }
+
+            } else {
+                // read-only
+                // connection.createStatement().execute("SET TRANSACTION READ ONLY");
+                connection.setAutoCommit(true);
+            }
```

Comment:
This entire `else` block can be removed if you move the `setAutoCommit(false)` statement above into the `if` block.